### ISSUE TITLE
Is it possible to add batch extraction of video subtitles？

### ISSUE THREAD

**Issue by Zero-net-oss** [Const-me/Whisper#153]
Is it possible to add batch extraction of video subtitles? If there are many video files, you need to wait in front of the computer all the time

**Comment by emcodem** [comment]
Mr. @tigros is your friend :D
https://github.com/tigros/Whisperer
Also, for a far more sophisticated automation solution, you can look at ffastrans.com. 

Note that i am co-developer at ffastrans and therefore as affiliated as it gets.

**Comment by Zero-net-oss** [comment]
> 先生。@tigros是你的朋友：D https://github.com/tigros/Whisperer 另外，对于更复杂的自动化解决方案，你可以查看 ffastrans.com。
> 
> 请注意，我是 ffastrans 的联合开发人员，因此与它有着密切的联系。

thank

**Comment by nandxorandor** [comment]
> Mr. @tigros is your friend :D https://github.com/tigros/Whisperer Also, for a far more sophisticated automation solution, you can look at ffastrans.com.
> 
> Note that i am co-developer at ffastrans and therefore as affiliated as it gets.

Hi Emcodem,

Is the whisperer an offline tool? Is it safe? Like absolutely no data leak? What would you recommend using it while connected to wifi or not connected?
Just to be honest, this is not only about whisperer, it is also about whisper, text-gen Webui, and many many other offline AI tools!
I would highly appreciate your input/thoughts!
Thanks

**Comment by emcodem** [comment]
@nandxorandor 
One can never know if any ready built program is safe to use for your unique security requirements (i wrote also here about it: https://github.com/Const-me/Whisper/issues/26#issuecomment-1711849399)
Basically @tigros Whisperer appears to me like it should be completely offline but without checking/measuring what the program does while execution, you cannot be sure (maybe not even the creator knows because he is using a shitload of dependencies like all programs that are written in so called high level programming languages like C# does).

To take it down to earth, there are lots of security measures applied before you can even use the program: Github checks the exe before you download, your browser checks it after it is downloaded but before you can use it and your antivirus checks it whenever you execute it. If that is not enogh for you, the only thing you can do is to measure what the tool does while execution in a sandbox environment using tools like e.g. procmon or similar.

**Comment by nandxorandor** [comment]
> @nandxorandor One can never know if any ready built program is safe to use for your unique security requirements (i wrote also here about it: [#26 (comment)](https://github.com/Const-me/Whisper/issues/26#issuecomment-1711849399)) Basically @tigros Whisperer appears to me like it should be completely offline but without checking/measuring what the program does while execution, you cannot be sure (maybe not even the creator knows because he is using a shitload of dependencies like all programs that are written in so called high level programming languages like C# does).
> 
> To take it down to earth, there are lots of security measures applied before you can even use the program: Github checks the exe before you download, your browser checks it after it is downloaded but before you can use it and your antivirus checks it whenever you execute it. If that is not enogh for you, the only thing you can do is to measure what the tool does while execution in a sandbox environment using tools like e.g. procmon or similar.

Thank you so much for those detailed info. 
It interesting that you mentioned to check with my Antivirus ... because when I wanted to run the .exe file windows gave me the "Windows protected you PC - Run Anyway - Don't Run" warning window, but I just ignored it and run it!!!! mmmmmmm too late I guess LoL

**Comment by emcodem** [comment]
@nandxorandor what you describe sounds very much like smartscreen blocking. In that case it is most likely that it is more like a setting on your PC than a real "virus detection". E.g. files are "blocked" when downloaded from the internet and not having a costly signed certificate. 
You can just unblock the file in the file properties window to disable this message:
https://www.advancedinstaller.com/prevent-smartscreen-from-appearing.html

**Comment by nandxorandor** [comment]
> @nandxorandor what you describe sounds very much like smartscreen blocking. In that case it is most likely that it is more like a setting on your PC than a real "virus detection". E.g. files are "blocked" when downloaded from the internet and not having a costly signed certificate. You can just unblock the file in the file properties window to disable this message: https://www.advancedinstaller.com/prevent-smartscreen-from-appearing.html

Yup! That's the one. Okay then, that's relieving if it is not serious or risky thing. Thanks again for taking time and effort to reply to my inquiries, I really appreciate that.
